src/backup.sh

```diff
@@ -0,0 +1,399 @@
+# This file deals with backing up and restoring kw data files stored under the
+# the KW_DATA_DIR directory. These files are configs, pomodoro reports and
+# kw usage statistics.
+
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kw_string.sh"
+include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_time_and_date.sh"
+
+declare -gA options_values
+decompress_path="$KW_CACHE_DIR/tmp-kw-backup"
+
+# This is the main function that deals with the operations related to backup. It
+# recieves an argument and call other functions based on that.
+function backup()
+{
+  parse_backup_options "$@"
+
+  if [[ -n "${options_values['BACKUP_PATH']}" ]]; then
+    create_backup "${options_values['BACKUP_PATH']}"
+    return
+  fi
+
+  if [[ -n "${options_values['RESTORE_PATH']}" ]]; then
+    restore_backup "${options_values['RESTORE_PATH']}"
+    return
+  fi
+}
+
+# This function creates a .tar.gz file containing all the data found in
+# KW_DATA_DIR
+#
+# @path Path to the directory in which the compressed file should be stored.
+function create_backup()
+{
+  local path="$1"
+  local flag="$2"
+  local file_name
+  local current_time
+  local ret
+  local compress_cmd
+
+  flag=${flag:-'SILENT'}
+
+  if [[ ! -d "$path" ]]; then
+    complain 'We could not find the path'
+    exit 2 # ENOENT
+  fi
+
+  current_time=$(date +%Y-%m-%d_%H-%M-%S)
+  file_name="kw-backup-$current_time.tar.gz"
+  compress_cmd="tar -C $KW_DATA_DIR -czf $path/$file_name ."
+  cmd_manager "$flag" "$compress_cmd"
+
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain 'We could not create the tar file'
+    exit "$ret"
+  fi
+
+  success 'Backup successfully created at' "$path/$file_name"
+}
+
+# This function restores a previous backup by extracting it back into
+# KW_DATA_DIR.
+#
+# #path Path to the .tar.gz containing kw data
+function restore_backup()
+{
+  local path="$1"
+  local flag="$2"
+  local ret
+  local decompress_cmd
+  local config_file
+  local difference
+
+  flag=${flag:-'SILENT'}
+
+  if [[ ! -f "$path" ]]; then
+    complain 'We could not find this file'
+    exit 2 # ENOENT
+  fi
+
+  if [[ -n "${options_values['FORCE']}" ]]; then
+    decompress_path="$KW_DATA_DIR"
+  else
+    mkdir -p "$decompress_path"
+  fi
+  decompress_cmd="tar -xzf $path -C $decompress_path"
+  cmd_manager "$flag" "$decompress_cmd"
+
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain 'We could not extract the tar file'
+    exit "$ret"
+  fi
+
+  if [[ -z "${options_values['FORCE']}" ]]; then
+    restore_config
+    restore_pomodoro_and_statistics
```

Comment:
I suppose we want to drop this in favor of restore_pomodoro and restore_statistics.